crates/metaslang/bindings/src/lib.rs

```diff
@@ -398,10 +456,11 @@ impl<'a, KT: KindTypes + 'static> Reference<'a, KT> {
         self.owner.cursors.get(&self.handle).cloned()
     }
 
-    pub fn get_file(&self) -> Option<&'a str> {
+    pub fn get_file(&self) -> FileDescriptor {
         self.owner.stack_graph[self.handle]
             .file()
-            .map(|file| self.owner.stack_graph[file].name())
+            .and_then(|file| FileDescriptor::from_string(self.owner.stack_graph[file].name()).ok())
+            .unwrap_or_else(|| unreachable!("Reference does not have a valid file descriptor"))
     }
```

Comment:
nit: Is `from_string()` ever expected to fail here? if not, I suggest propagating the error instead of silencing it here:

```suggestion
            .and_then(|file| FileDescriptor::from_string(self.owner.stack_graph[file].name()))
            .expect("Reference does not have a valid file descriptor")
    }
```